### PR TITLE
Add User Create API and Test it with Json Passing edge cases

### PR DESCRIPTION
**Pull request by Sourav1343**
## Summary by CodeRabbit

- **New Features**
	- Introduced a RESTful API endpoint for user creation, allowing users to register through a structured process.
	- Added functionality to validate user input and handle errors during the registration process.
	- Implemented a mechanism to prevent duplicate email registrations.

- **Bug Fixes**
	- Improved error handling for user registration, ensuring appropriate responses for validation and database errors.

- **Refactor**
	- Streamlined the user repository interface by removing unnecessary custom query methods, enhancing code simplicity and maintainability.

### CODE REVIEW

File: src/main/java/com/portal/attendance/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private UserRepository userRepository;
+
+
+    @Override
+    public User saveUser(User user) {
+        // Check if email is already present in the database
+        Optional<User> existingUser = userRepository.findByEmail(user.getEmail());
+        if (existingUser.isPresent()) {
+            throw new IllegalArgumentException("User with this email already exists");
+        }
```
[the remark this answers is not in the document]
Comment (Sourav1343) [reply src/main/java/com/portal/attendance/service/impl/UserServiceImpl.java]: DONE

[the remark this answers is not in the document]
Comment (Sourav1343) [reply src/main/java/com/portal/attendance/service/impl/UserServiceImpl.java]: DONE
